lib/runctl.js

```diff
@@ -49,6 +49,13 @@ if (cluster.isMaster && !process.env.SL_RUN_SKIP_IPCCTL) {
   if (wsUrl) {
     debug('runctl connect: %s', wsUrl);
     ipcctl = wsChannel(onRequest, wsUrl);
+    ipcctl.on('error', function(err) {
+      // TODO: throttle reconnects, eventually give up?
+      console.error('supervisor lost connection to contorl channel: ', err);
+      ipcctl.connect(wsUrl);
```

Comment:
@piscisaureus connect on error should be unnecessary after the changes you are working on, right?

And unref should persist across new incoming ws connections.